gradio/routes.py

```diff
@@ -588,29 +588,26 @@ def robots_txt():
 ########
 
 
-def is_relative_to(self, other):
-    # method exist on PurePath class from 3.9
-    try:
-        self.relative_to(other)
-        return True
-    except ValueError:
-        return False
-
-
-def safe_join(directory: str, path: str) -> str:
-    if path == "":
-        raise HTTPException(400)
-
-    filename = Path(path)
-    root = Path(directory)
-    full_path = root.joinpath(filename).resolve()
+def safe_join(directory: str, path: str) -> str | None:
```

Comment:
Not sure why you rolled back that change. It will break other part of the code which now rely on this.